### PR TITLE
Issue #40 Add validation rules for config file

### PR DESCRIPTION
**Pull request by JayjeetAtGithub**
Signed-off-by: Jayjeet Chakraborty <jc.github@rediffmail.com>

### CODE REVIEW

**Comment by JayjeetAtGithub** [comment]
@Harkishen-Singh @muskankhedia Not sure why the build is failing. Can you please check ?
Also, Please review my PR. Thanks.

**Comment by muskankhedia** [comment]
@JayjeetAtGithub Nice Work! Some minor changes required, please try to create a new module for checking the `local-config file` as created for `parsing` or `ping` as it will help to maintain the code and different modules for different features or checks will be defined.

**Comment by JayjeetAtGithub** [comment]
Ok ! Will do it.

**Comment by JayjeetAtGithub** [comment]
@muskankhedia Can we just create a `config_validator.go` in the `parser` package ? This way we can keep all the parsing related logic in the same pkg.

**Comment by muskankhedia** [comment]
@JayjeetAtGithub Yaa sure. Sounds better

**Comment by JayjeetAtGithub** [comment]
> @muskankhedia Can we just create a `config_validator.go` in the `parser` package ? This way we can keep all the parsing related logic in the same pkg.

@muskankhedia  I updated the PR with this change. Please review. Thanks.

File: src/lib/utils/parser/config-validator.go [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+			if len(route.Header) != 0 {
+				for j, header := range route.Header {
+					if header.OfType == "" {
+						log.Fatalf("`type` property for header #%d of route `%s` is missing.", j+1, route.Route)
+					}
+					if header.Value == "" {
+						log.Fatalf("`value` property for header #%d of route `%s` is missing.", j+1, route.Route)
+					}
+				}
+			}
```
Comment (muskankhedia) [inline src/lib/utils/parser/config-validator.go]: @JayjeetAtGithub For now, I think we don't need to make `header` field mandatory. Can you please change it for now?

File: src/lib/utils/parser/config-validator.go [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+			if len(route.Params) != 0 {
+				for j, param := range route.Params {
+					if param.Name == "" {
+						log.Fatalf("`name` property for param #%d of route `%s` is missing.", j+1, route.Route)
+					}
+					if param.Value == "" {
+						log.Fatalf("`value` property for param #%d of route `%s` is missing.", j+1, route.Route)
+					}
+				}
+			}
```
Comment (muskankhedia) [inline src/lib/utils/parser/config-validator.go]: Same goes for this. Please have a look and implement the same.

**Comment by Harkishen-Singh** [comment]
@JayjeetAtGithub any updates?

**Comment by JayjeetAtGithub** [comment]
@Harkishen-Singh Sorry for the gap. I will get back to it tomorrow. Does that work ?

**Comment by Harkishen-Singh** [comment]
@JayjeetAtGithub perfect!

**Comment by JayjeetAtGithub** [comment]
Thanks

File: src/lib/utils/parser/config-validator.go (hunk 1 above)
Comment (JayjeetAtGithub) [reply src/lib/utils/parser/config-validator.go]: I just checked and realized that `header` field is not mandatory in the above code. I am checking for validation of headers only if it is present.

File: src/lib/utils/parser/config-validator.go (hunk 2 above)
Comment (JayjeetAtGithub) [reply src/lib/utils/parser/config-validator.go]: same for `params`.

**Comment by JayjeetAtGithub** [comment]
@muskankhedia  looks like the requested changes are already implemented. Can you please double check ? Thanks a lot.

**Comment by muskankhedia** [comment]
@JayjeetAtGithub Lgtm, @Harkishen-Singh Please have a look.

**Pull request merged**